src/CLI/Options.cs

```diff
@@ -16,52 +11,36 @@ enum TargetArchitecture
 
     class Options
     {
-        private List<String> _headerFiles = new List<string>();
-        public List<String> HeaderFiles { get { return _headerFiles; } set { _headerFiles = value; } }
+        public List<string> HeaderFiles { get; } = new List<string>();
 
-        private List<String> _includeDirs = new List<string>();
-        public List<String> IncludeDirs { get { return _includeDirs; } set { _includeDirs = value; } }
+        public List<string> IncludeDirs { get; } = new List<string>();
 
-        private List<String> _libraryDirs = new List<string>();
-        public List<String> LibraryDirs { get { return _libraryDirs; } set { _libraryDirs = value; } }
+        public List<string> LibraryDirs { get; } = new List<string>();
 
-        private List<String> _libraries = new List<string>();
-        public List<String> Libraries { get { return _libraries; } set { _libraries = value; } }
+        public List<string> Libraries { get; } = new List<string>();
 
-        private Dictionary<String, String> _defines = new Dictionary<String, String>();
-        public Dictionary<String, String> Defines { get { return _defines; } set { _defines = value; } }
+        public Dictionary<string, string> Defines { get; } = new Dictionary<string, string>();
 
-        private String _outputDir = "";
-        public String OutputDir { get { return _outputDir; } set { _outputDir = value; } }
+        public string OutputDir { get; set; } = "";
```

Comment:
Now that you use string.IsNullOrEmpty, this initialisation is needless, the default value of "null" will work just the same:

public string OutputDir { get; set; };